OpenSim/Actuators/VandenBogert2011Muscle.cpp

```diff
@@ -0,0 +1,1234 @@
+        //      and fiberLength ~ projFiberLength
+        cosPenn=1.0;
+        fiberLengthNorm=projFibLenNorm;
+        dfiberLength_dprojFibLen=1;
+        dcosPenn_dprojFibLen=0;
+    }
+    else {
+        double b=sin(pennAtOptFiberLength);
+        fiberLengthNorm=sqrt(pow(projFibLenNorm,2) + pow(b,2));
+        cosPenn = projFibLenNorm/fiberLengthNorm;
+        dfiberLength_dprojFibLen=cosPenn;
+        dcosPenn_dprojFibLen=pow(b,2) / pow(fiberLengthNorm,3);
+
+    }
+
+    // Compute fiberVelocity and its derivatives wrt projFibLen and projFibVel
+    double fiberVelocityNorm = projFibVelNorm*cosPenn;
+    double dfiberVelocityNorm_dprojFibVelNorm = cosPenn;
+    double dfiberVelocityNorm_dprojFibLenNorm =
+            projFibVelNorm * dcosPenn_dprojFibLen;
+
+
+    //---F1 is the normalized isometric force-length relationship at maximum
+    //                                               activation--------------//
+    double fiberExp = (fiberLengthNorm - 1.0) / fl_width;   // [dimensionless]
+    double F1 = exp(-pow(fiberExp, 2));        // Gaussian force-length curve
+
+    double dF1_dfiberLengthNorm = 0;
+    if (returnJacobians) {
+        dF1_dfiberLengthNorm = -2.0 * fiberExp * F1 / fl_width;
+        double dF1_dprojFibLenNorm =
+                dF1_dfiberLengthNorm * dfiberLength_dprojFibLen;
+    }
+
+
+
+    //-------- F2 is the dimensionless force-velocity relationship -------//
+    double F2;
+    double dF2_dfiberVelocityNorm;
+    double dF2_dactiv;
+    double dF2_dprojFibVelNorm;
+    double dF2_dprojFibLenNorm;
+    double df_dmuscleLength;
+
+    // Chow/Darling Vel-Activation Relationship //TODO:  Add full reference
+    //double lambda = 0.5025 + 0.5341*activ;
+    //double  dlambda_da = 0.5341;
+    double lambda = 1;   //Turn it off for now as it seems to cause an issue
+                        // with negative force large concentric vel
+    double dlambda_da =0;
+
+    if (fiberVelocityNorm < 0) {
+        //Hill's equation for concentric contraction
+        // F2 = (V_{max} + V_{fiber}) / (V_{max} - V_{fiber}/a_{Hill})
+
+        double hillDenom = (lambda*vMaxNorm - fiberVelocityNorm/fv_AHill);
+        F2 = (lambda*vMaxNorm + fiberVelocityNorm) / hillDenom;
+
+        if (returnJacobians) {
+            dF2_dfiberVelocityNorm  = (1.0 + F2 / fv_AHill) / hillDenom;
+            dF2_dactiv = - dlambda_da * vMaxNorm * fiberVelocityNorm *
+                    (1.0 + 1.0/fv_AHill) / pow(hillDenom,2);
+        }
+    }
+    else {
+        //Katz Curve for eccentric contraction
+        // c is Katz Constant
+        double c3 = vMaxNorm * fv_AHill * (fv_maxMultiplier - 1.0) /
+                (fv_AHill + 1.0); // parameter in the eccentric f-v equation
+        double c = lambda*c3;
+        //F2 = (g_{max} * V_{fiber} + c) / (V_{fiber} + c)
+        double katzDenom = (fiberVelocityNorm  + c);
+        F2 = (fv_maxMultiplier * fiberVelocityNorm  + c) / katzDenom ;
```

Comment:
I plotted this and I noticed that the maximum value of the force is 1.69-1.74 (depending on the value of `lambda`, or `activation`) when `fv_maxMultiplier = 1.8`, within the range `-10 <= v <= 10` (fiber lengths per second). As expected, the f-v curve *does* asymptote to 1.8 as `v -> inf`. Do you plan to limit the max eccentric velocity according to the `max_contraction_velocity` parameter? If so, I somewhat expect the `fv_maxMultiplier` to be somewhat close to the max value of F2 at `v = 10`. I didn't look into what other muscles do; perhaps this is not an issue.